### PR TITLE
CB-15758 Better way to check if the nodegroup is only for compute for…

### PR DESCRIPTION
**Pull request by cegganesh84**
… recommissioning during stop start scaling cluster start

1. Re-used the same method used in autoscaling enablement for figuring out compute nodes, instead of relying on the name compute.
2. In ClusterStartHandler it uses the hostname pattern (<cluster-name>-<group-name><num>) to identify if it is compute hostname or not.
3. Added unit tests to make sure both the status syncer and cluster start works according to the expectation.

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java
Comment (sidseth) [inline core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java:365]: Wouldn't the keySet include all hostGroups containing YARN? The filtering needs to be based on the values and the attributes set there?

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java
Comment (sidseth) [inline core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java:156]: Same comment as above on the keySet not being adequate. FWIW: Java streams, IMO, makes code more difficult to read / get right.

File: core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java
Comment (cegganesh84) [reply core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java]: Great catch, fixed it.

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java
Comment (cegganesh84) [reply core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java]: Unrolled the streams.

File: core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java
Comment (sidseth) [inline core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java:152]: Nit: Separating this check can avoid a call to CM if there are not compute groups.

**Review by sidseth — APPROVED** [review]
Looks good. Posted 1 comment, which I'll address.

Holding off committing this patch to CB-14929 since that is under active review. Once that is merged, will change this PR to be against master.

**Comment by sidseth** [comment]
Will need to be rebased.

**Pull request merged**
